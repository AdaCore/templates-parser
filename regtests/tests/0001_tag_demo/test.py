from test_support import *

gnatmake('tag_demo')
run('tag_demo')
