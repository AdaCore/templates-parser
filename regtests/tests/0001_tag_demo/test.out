T1
 (N= 2, Min= 1, Max= 1, Nested_Level= 1
  first
  1_1_1
 )
T2
 (N= 1, Min= 1, Max= 1, Nested_Level= 1
  second
 )
T
 (N= 2, Min= 1, Max= 2, Nested_Level= 2
  (N= 2, Min= 1, Max= 1, Nested_Level= 1
    first
    1_1_1
  )
  (N= 1, Min= 1, Max= 1, Nested_Level= 1
    second
  )
 )
I3
 (N= 3, Min= 1, Max= 1, Nested_Level= 1
  3
  2
  1
 )
M
 (N= 3, Min= 2, Max= 2, Nested_Level= 3
  (N= 2, Min= 1, Max= 2, Nested_Level= 2
   (N= 2, Min= 1, Max= 1, Nested_Level= 1
      first
      1_1_1
   )
   (N= 1, Min= 1, Max= 1, Nested_Level= 1
      second
   )
  )
  (N= 2, Min= 1, Max= 2, Nested_Level= 2
   (N= 2, Min= 1, Max= 1, Nested_Level= 1
      first
      1_1_1
   )
   (N= 1, Min= 1, Max= 1, Nested_Level= 1
      second
   )
  )
  (N= 2, Min= 1, Max= 2, Nested_Level= 2
   (N= 2, Min= 1, Max= 1, Nested_Level= 1
      first
      1_1_1
   )
   (N= 1, Min= 1, Max= 1, Nested_Level= 1
      second
   )
  )
 )

GTAG

----------------------------
0 : first, 1_1_1
second

----------------------------
1 : first, 1_1_1
1 : second

----------------------------
2 : first
2 : 1_1_1
2 : second
2 : 

----------------------------
M0 : first, 1_1_1
second

M1 : first, 1_1_1
M1 : second

M2 : first
M2 : 1_1_1
M2 : second
M2 : 

============================

VTAG

----------------------------
0 : first, 1_1_1

----------------------------
1 : first
1 : 1_1_1

----------------------------
[
V2 : first
2  : first
V2 : 1_1_1
2  : 1_1_1
]
[
V2 : first
2  : second
V2 : 1_1_1
2  : 
]

[
O2 : first
M2 : first
O2 : 1_1_1
M2 : 1_1_1
]
[
O2 : first
M2 : second
O2 : 1_1_1
M2 : 
]
----------------------------
-----------------------
T1=>("first","1_1_1")
T1  ("first","1_1_1")
   Ok
T2=>("second")
T2  ("second")
   Ok
T =>(("first","1_1_1")("second"))
T   (("first","1_1_1")("second"))
   Ok
T=>(("first","1_1_1")("second")("un essai","with "" a quote","and ( and )"))
T  (("first","1_1_1")("second")("un essai","with "" a quote","and ( and )"))
   Ok
